### PR TITLE
drivers/sensor: add support to IIS2DLPC accelerometer sensor

### PR DESCRIPTION
**Pull request by avisconti**
The IIS2DLPC is a 3D digital accelerometer ultra-low power sensor
for industrial applications, which can be interfaced through either
I2C or SPI bus.

https://www.st.com/resource/en/datasheet/iis2dlpc.pdf

This driver is based on stmemsc i/f v1.02

NOTE::
Driver tested using x_nucleo_iks02a1 (not yet submitted)

### CODE REVIEW

**Comment by avisconti** [comment]
Maybe better to add the test to build it

File: drivers/sensor/iis2dlpc/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# ST Microelectronics IIS2DLPC 3-axis accelerometer driver
+#
+# Copyright (c) 2019 STMicroelectronics
```
Comment (MaureenHelm) [inline drivers/sensor/iis2dlpc/CMakeLists.txt]: Update year to 2020

File: drivers/sensor/iis2dlpc/iis2dlpc.h [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	u8_t pulse_ths[3];
+	u8_t pulse_shock;
+	u8_t pulse_ltncy;
+	u8_t pulse_quiet;
+#endif /* CONFIG_IIS2DLPC_PULSE */
+#endif /* CONFIG_IIS2DLPC_TRIGGER */
+};
+
+/* sensor data forward declaration (member definition is below) */
+struct iis2dlpc_data;
```
Comment (MaureenHelm) [inline drivers/sensor/iis2dlpc/iis2dlpc.h]: Is this actually necessary?

File: dts/bindings/sensor/st,iis2dlpc-i2c.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2018 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    STMicroelectronics IIS2DLPC magnetometer accessed through I2C bus
```
Comment (MaureenHelm) [inline dts/bindings/sensor/st,iis2dlpc-i2c.yaml]: accelerometer

File: dts/bindings/sensor/st,iis2dlpc-spi.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2019 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    STMicroelectronics IIS2DLPC magnetometer accessed through SPI bus
```
Comment (MaureenHelm) [inline dts/bindings/sensor/st,iis2dlpc-spi.yaml]: accelerometer

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: dts/bindings/sensor/st,iis2dlpc-i2c.yaml (hunk 3 above)
Comment (avisconti) [reply dts/bindings/sensor/st,iis2dlpc-i2c.yaml]: ack

File: dts/bindings/sensor/st,iis2dlpc-spi.yaml (hunk 4 above)
Comment (avisconti) [reply dts/bindings/sensor/st,iis2dlpc-spi.yaml]: ack

File: drivers/sensor/iis2dlpc/CMakeLists.txt (hunk 1 above)
Comment (avisconti) [reply drivers/sensor/iis2dlpc/CMakeLists.txt]: sure. Thanks! :(

File: drivers/sensor/iis2dlpc/iis2dlpc.h (hunk 2 above)
Comment (avisconti) [reply drivers/sensor/iis2dlpc/iis2dlpc.h]: Of course not. It probably remained there by mistake.

**Review by MaureenHelm — APPROVED** [review]

**Comment by avisconti** [comment]
@erwango 
Any feedback?

**Comment by jhedberg** [comment]
@erwango do you want to take a look at this one before it gets merged? Btw, I just triggered a re-run of Shippable since it was a month ago when it last happened.

**Review by erwango — APPROVED** [review]

**Pull request merged**